### DISCUSSION TITLE
How to migrate from nginx to apisix gracefully

### DISCUSSION THREAD

**Discussion by 1393467261** [apache/apisix#5765] [Q&A] [answered]
For the last serveral months I've been doing the migrating things for 1000+ domains in stage. Apisix is great, but I got into trouble where the client using `http` will receive `301|308` while accessing domain allowing `http to https`.It's impossible to modify the config(using https or following redirect) of client because there are too many of them. Is Apisix support both http and https for a single domain without redirect like nginx?

**Comment by juzhiyuan** [comment]
Hi @tokers @bzp2010 @tzssangglass, do you have ideas on this? 😄

**Reply by tokers** [reply]
Did you enable the `redirect` plugin and restrict the `HTTP` protocol? By default, APISIX won't redirect `HTTP` requests (to `HTTPS`).

**Comment by bzp2010** [comment, marked as answer]
As far as I know, APISIX will not force a 301 jump to HTTPS unless you have the `redirect` plugin enabled.

**Reply by 1393467261** [reply]
I have to enable `redirect` because there are some domains both need `https` and `http` for some historical reasons. Is there any other way to   
 achive it?

**Reply by tokers** [reply]
I'm confused, if you want to enable both the HTTP and HTTPS for some domains, you don't have to enable this plugin.

**Reply by 1393467261** [reply]
Oh, I see. That's it. Thanks you!

**Reply by juzhiyuan** [reply]
Hi @1393467261, today we have an online meeting, welcome to join, and here are the details: https://github.com/apache/apisix/discussions/5786

You could find topics here: https://docs.qq.com/doc/DTkhKZFJtVHB5VXZ1

**Reply by 1393467261** [reply]
@juzhiyuan Thanks for the invitation. So sorry I haven't check the email recently, what's a pity. Looking forward to further cooperation.

**Reply by juzhiyuan** [reply]
Hi @1393467261, we have the coming meeting tomorrow evening:) See https://docs.qq.com/doc/DTktRakxqaUFFWndz
